### ISSUE TITLE
[Screen Reader] Project card selection state (selected/unselected) is not announced on “View All My Projects” page during keyboard navigation

### ISSUE THREAD

**Issue by Nina-1210** [microsoft/pxt-microbit#6979]
**Describe the bug**
When using a screen reader (Windows Narrator or macOS VoiceOver), the selection state of project cards (represented by the selected/unselected circle icon) is not announced on the “View All My Projects” page during keyboard navigation.

**Steps to reproduce the behavior:**
1. Enable a screen reader:
    -  Windows Narrator (Windows 11) or
    -  VoiceOver (macOS)
2. Navigate to [Microsoft MakeCode for micro:bit](https://makecode.microbit.org/beta).
3. Tab to the “**View All**” button and press **Enter**.
4. Open the “**View All My Projects**” page.
5. Continue pressing **Tab** until focus reaches a project card.
6. Press **Enter** repeatedly to toggle selection.
7. Observe the screen reader output.

**Actual behavior**
The screen reader does not announce any change in selection state (selected or unselected) for the project card.
 
**Expected behavior**
The screen reader should announce the correct selection state (e.g., “selected” / “not selected”) when the project card is focused and toggled.

**Screen Recording**
https://github.com/user-attachments/assets/48cf9f56-9477-477e-a01b-f8f33580783e

**Additional context**
1.OS: Windows 11 25H2 & Mac Mini M2 macOS Tahoe 26.5.1
2.Browser: Chrome, Edge, Firefox, and Safari
3.makecode.microbit.org version: 8.1.28
4.Microsoft MakeCode version: 12.3.24
5.Screen Readers: Narrator and VoiceOver

**Comment by abchatra** [comment]
@microbit-robert

**Comment by microbit-robert** [comment]
These project cards are currently buttons, each with an additional label inside acting as the visual representation as to whether the card is currently selected or not. This label has a role of button, and is unreachable by keyboard navigation. You can use keyboard navigation to toggle selection state when the card itself (not the label acting as the circular checkbox) has focus.

These cards only serve one purpose, to be selected or not, and any interaction with them toggles this state. They are likely better represented to screen reader users as checkboxes which will then naturally have a selected state. 

This would involved non-trivial changes to the code card component which is leveraged extensively in different parts of the code base. We feel the MakeCode team are best place to consider the design and implementation of this fix.
